frontend/src/metabase/containers/AddToDashSelectDashModal/AddToDashSelectDashModal.tsx

```diff
@@ -35,26 +34,38 @@ interface AddToDashSelectDashModalProps {
 
 type DashboardPickerProps = ComponentPropsWithoutRef<typeof DashboardPicker>;
 
-export const AddToDashSelectDashModal = ({
+const AddToDashSelectDashModal = ({
   card,
   dashboards,
   onClose,
   onChangeLocation,
 }: AddToDashSelectDashModalProps) => {
   const [shouldCreateDashboard, setShouldCreateDashboard] = useState(false);
 
-  const mostRecentDashboardQuery = useMostRecentlyViewedDashboard();
-
-  const collectionId = coerceCollectionId(
-    mostRecentDashboardQuery.data?.collection_id,
-  );
+  const mostRecentlyViewedDashboardQuery = useMostRecentlyViewedDashboard();
+  const mostRecentlyViewedDashboard = mostRecentlyViewedDashboardQuery.data;
+  const isQuestionInPersonalCollection = Boolean(card.collection?.is_personal);
+  const collectionId = useCollectionId({
+    isQuestionInPersonalCollection,
+    mostRecentlyViewedDashboard,
+  });
   // when collectionId is null and loading is completed, show root collection
-  // as user didnt' visit any dashboard last 24hrs
+  // as user didn't visit any dashboard last 24hrs
   const collectionQuery = useCollectionQuery({
     id: collectionId,
     enabled: collectionId !== undefined,
   });
 
+  const [openCollectionId, setOpenCollectionId] = useState<
+    CollectionId | undefined
+  >();
+  const openCollection = useSelector(state =>
+    getCollection(state, { collectionId: openCollectionId }),
+  );
+  const isOpenCollectionInPersonalCollection = openCollection?.is_personal;
+  const hideCreateNewDashboardOption =
```

Comment:
This only happens at the root collection. Because public collections will already be hidden when selecting a dashboard for a question in a personal collection.